sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -904,6 +929,8 @@ public static class Configuration {
         private Boolean commitOffsetInFinalize;
         private Long consumerPollingTimeout;
         private String timestampPolicy;
+        private Long redistributeNumShards;
```

Comment:
When you pass this to Redistribute you cast it to an `int`. So just use `Integer` here if it has to be that small (which is obviously plenty big since the goal is to reduce the number of keys).

Actually we should rename all of this to `numKeys` because "shard" is an element of processing and it is a Dataflow-specific quirk that key == shard even when it is not required.

Reply:
Done